python/foglamp/plugins/south/systeminfo/systeminfo.py

```diff
@@ -0,0 +1,322 @@
+                col_heads[3]: col_vals[3],
+                col_heads[4]: col_vals[4],
+                col_heads[5]: col_vals[5],
+            })
+        data.update({
+            "diskUsage": disk_usage
+        })
+
+        no_of_processes = get_subprocess_result(cmd='ps -eaf | wc -l')[0]
+        data.update({
+            "numberOProcessesRunning": no_of_processes
+        })
+
+        # Get Network and other info
+        data.update({
+            "networkTraffic": get_network_traffic()
+        })
+
+        # Paging and Swapping
+        c6 = get_subprocess_result(cmd='vmstat -s')
+        paging_swapping = []
+        for line in c6:
+            if 'page' in line:
+                paging_swapping.append(line)
+        data.update({
+            "numberOfPagingAndSwappingEvents": paging_swapping
+        })
+
+        # Disk Traffic
+        c4 = get_subprocess_result(cmd='iostat -xd 2 1')
+        c5 = c4[2:]
+        disk_traffic = []
+        col_heads = c5[0].split()
+        for line in c5[1:]:
+            if line == '':
+                continue
+            col_vals = line.split()
+            disk_traffic.append({
+                col_heads[0]: col_vals[0],
+                col_heads[1]: col_vals[1],
+                col_heads[2]: col_vals[2],
+                col_heads[3]: col_vals[3],
+                col_heads[4]: col_vals[4],
+                col_heads[5]: col_vals[5],
+                col_heads[6]: col_vals[6],
+                col_heads[7]: col_vals[7],
+                col_heads[8]: col_vals[8],
+                col_heads[9]: col_vals[9],
+                col_heads[10]: col_vals[10],
+                col_heads[11]: col_vals[11],
+                col_heads[12]: col_vals[12],
+                col_heads[13]: col_vals[13]
+            })
+        data.update({
+            "platform": c4[0],
+            "diskTraffic": disk_traffic
+        })
+        return data
+
+    async def save_data():
+        try:
+            while True:
+                time_stamp = utils.local_timestamp()
+                data = {
+                    'asset': 'systeminfo',
+                    'timestamp': time_stamp,
+                    'key': str(uuid.uuid4()),
+                    'readings': {
+                        handle['assetCode']['value']: get_system_info()
+                    }
+                }
+
+                await Ingest.add_readings(asset='{}'.format(data['asset']),
+                                          timestamp=data['timestamp'], key=data['key'],
+                                          readings=data['readings'])
+
+                await asyncio.sleep(int(handle['sleepInterval']['value']))
+
+        except asyncio.CancelledError:
+            pass
+
+        except (Exception, RuntimeError) as ex:
+            _LOGGER.exception("System Info exception: {}".format(str(ex)))
+            raise exceptions.DataRetrievalError(ex)
+
+    asyncio.ensure_future(save_data())
```

Comment:
except/ else is more pythonic

Reply:
Nope. "asyncio.ensure_future(save_data())" is not part of above code segment.

Reply:
ah! Okay